docs/query-mode.md

```diff
@@ -0,0 +1,51 @@
+The Sarif.Multitool has a 'query' mode. Query mode takes a SarifLog and query expression and determines how many Results match the expression. It can write the count to the console, return the count as an exit code, signal if the count exceeds a threshold, or write an output log with the filtered results only.
+
+Query mode is not as sophisticated as the SARIF viewers, but can be used to separate logs which are too large into meaningful subsets or to trigger on simple conditions as part of an automated workflow. For arbitrarily complex scenarios, it's easiest to reference the Sarif SDK and write the logic as code.
+
+### Examples
+
+| Command | Meaning |
+| ------- | ------- |
+| Sarif.Multitool query In.sarif -e "RuleId = SM00251"    | Output Result count with RuleId SM00251 |
+| Sarif.Multitool query In.sarif -e "RuleId = SM00251" -o Out.sarif | ...and write filtered log to 'Out.sarif' |
+| ... -e "Uri : 'Test Framework'" | Find Results where any URI contains 'Test Framework' |
+| ... -e "!(Uri >&#124;> .cs)"   | Find Results where no URI EndWith '.cs' |
+| ... -e "OccurrenceCount > 10 AND RuleId = SM00251" | Look for Results which match both criteria |
+
+
+### Comparison Operators
```

Comment:
>Operators [](start = 15, length = 9)

Is there a precedence hierarchy? If so, this table should be in hierarchical order (and should say so). If there's no hierarchy, we should _really_ say so :-)

Reply:
Very good suggestion. They are: NOT, AND, OR. I've ordered them and added an example.

Reply:
And the other table, with `==`, `<`, etc -- does that also have a hierarchy? I think we need a single table with the entire hierarchy.

---
In reply to: [375554889](https://github.com/microsoft/sarif-sdk/pull/1743#discussion_r375554889) [](ancestors = 375554889)